packages/fuels-programs/src/assembly/script_and_predicate_loader.rs

```diff
@@ -42,16 +43,17 @@ impl LoaderCode {
 
     #[cfg(feature = "std")]
     pub fn extract_blob(binary: &[u8]) -> Result<fuels_core::types::transaction_builders::Blob> {
-        let (code, _) = split_at_data_offset(binary)?;
+        let (code, _) = split_for_loader(binary)?;
         Ok(code.to_vec().into())
     }
 
     pub fn as_bytes(&self) -> &[u8] {
         &self.code
     }
 
+    #[deprecated]
     pub fn data_section_offset(&self) -> usize {
-        self.data_offset
+        self.section_offset
     }
 
     fn generate_loader_code(blob_id: [u8; 32], data_section: &[u8]) -> (Vec<u8>, usize) {
```

Comment:
could be either data or configurable section at this point. Would call it `split_section` or something else
```suggestion
    fn generate_loader_code(blob_id: [u8; 32], spli_section: &[u8]) -> (Vec<u8>, usize) {
```

Reply:
in this function there are helper functions that say `data_section` I assume we would have to change those also.

Reply:
I do not know the difference between `data` and `configurable` offfset, but now that we use the `configurable`and not `data` offset do we still have to have two different loader codes ?
`loader_instructions_w_data_section` and `loader_instructions_no_data_section`